qiskit_nature/problems/second_quantization/lattice/lattice/lattice.py

```diff
@@ -0,0 +1,243 @@
+from retworkx.visualization import mpl_draw
+
+
+@dataclass
+class DrawStyle:
+    """A stylesheet for lattice figure.
+    Please see
+    https://qiskit.org/documentation/retworkx/stubs/retworkx.visualization.mpl_draw.html#retworkx.visualization.mpl_draw
+    for each element.
+    """
+
+    # position
+    pos: Optional[dict] = None
+
+    # Matplotlib Axes object
+    ax: Optional[Axes] = None  # pylint:disable=invalid-name
+
+    with_labels: bool = False
+
+    node_list: Optional[list] = None
+
+    edge_list: Optional[list] = None
+
+    node_size: Union[int, list] = 300
+
+    node_color: Union[
+        str,
+        Tuple[float, float, float],
+        Tuple[float, float, float],
+        List[Union[str, Tuple[float, float, float], Tuple[float, float, float, float]]],
+    ] = "#1f78b4"
+
+    node_shape: str = "o"
+
+    # node and edge transparency
+    alpha: Optional[float] = None
+
+    # Matplotlib colormap object
+    cmap: Optional[Colormap] = None
+
+    # minimum value for node colormap scaling
+    vmin: Optional[float] = None
+
+    # minimum value for node colormap scaling
+    vmax: Optional[float] = None
+
+    linewidths: Union[float, Sequence] = 1.0
+
+    width: Union[float, Sequence] = 1.0
+
+    edge_color: Union[str, Sequence] = "k"
+
+    edge_cmap: Optional[Colormap] = None
+
+    # minimum value for edge colormap scaling
+    edge_vmin: Optional[float] = None
+
+    # maximum value for node colormap scaling
+    edge_vmax: Optional[float] = None
+
+    style: str = "solid"
+
+    labels: Optional[Callable] = None
+
+    edge_labels: Optional[Callable] = None
+
+    font_size: int = 12
+
+    font_color: str = "k"
+
+    font_weight: str = "normal"
+
+    font_family: str = "sans-serif"
+
+    label: Optional[str] = None
+
+    connectionstyle: str = "arc3"
+
+
+class Lattice:
+    """General Lattice."""
+
+    def __init__(self, graph: PyGraph) -> None:
+        """
+        Args:
+            graph: Input graph for Lattice. `graph.multigraph` must be False.
+
+        Raises:
+            ValueError: If `graph.multigraph` is True for a given graph, it is invalid.
+        """
+        if graph.multigraph:
+            raise ValueError(
+                f"Invalid `graph.multigraph` {graph.multigraph} is given. "
+                "`graph.multigraph` must be `False`."
+            )
+        if graph.edges() == [None] * graph.num_edges():
+            weighted_edges = [edge + (1.0,) for edge in graph.edge_list()]
+            for start, end, weight in weighted_edges:
+                graph.update_edge(start, end, weight)
+        self._graph = graph
+
+        self.pos: Optional[dict] = None
+
+    @property
+    def graph(self) -> PyGraph:
+        """Return a copy of the input graph."""
+        return self._graph.copy()
+
+    @property
+    def num_nodes(self) -> int:
+        """Return the number of nodes."""
+        return self.graph.num_nodes()
+
+    @property
+    def node_indexes(self) -> NodeIndices:
```

Comment:
out of curiosity is it indexes or indices? Because is a bit strange to have `node_indexes` with type `NodeIndices`.

Reply:
It's a retworkx quirk. The retworkx graph method is `.node_indexes()` which has been there since the earliest releases of retworkx. But it returns a `NodeIndices` object. I've thought about fixing the method name, but it is not worth the api breakage imo. I'd rather keep people's code working then fix a small wart in the retworkx api. That being said I think you can call the property here `node_indices` and avoid the issue retworkx has.